numpy_financial/_financial.py

```diff
@@ -840,20 +825,40 @@ def irr(values, *, guess=None, tol=1e-12, maxiter=100, raise_exceptions=False):
     #
     # which we solve using Newton-Raphson and then reverse out the solution
     # as eirr = g - 1 (if we are close enough to a solution)
-    npv_ = np.polynomial.Polynomial(values[::-1])
-    d_npv = npv_.deriv()
-    g = 1 + guess
+    
+    g = np.roots(values)
```

Comment:
From the [roots](https://numpy.org/doc/stable/reference/generated/numpy.roots.html) docs:

> This forms part of the old polynomial API. Since version 1.4, the new polynomial API defined in [numpy.polynomial](https://numpy.org/doc/stable/reference/routines.polynomials.package.html#module-numpy.polynomial) is preferred. A summary of the differences can be found in the [transition guide](https://numpy.org/doc/stable/reference/routines.polynomials.html).

However I'm not sure if it's worth to construct a whole ``Polynomial`` object just for this one calculation...

Reply:
I'm currently re-writing some of the functions to use [numba](https://numba.readthedocs.io/en/stable/index.html), it looks like ``np.roots`` is supported, but the new Polynomial API is not. So I'd prefer this version.